yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -756,7 +756,12 @@ private[spark] class Client(
     val interval = sparkConf.getLong("spark.yarn.report.interval", 1000)
     var lastState: YarnApplicationState = null
     while (true) {
-      Thread.sleep(interval)
+      try {
+        Thread.sleep(interval)
+      } catch {
+        case i: InterruptedException => // Swallow the InterruptedException
```

Comment:
Thanks for pointing out, this actually might be a problem after revisiting the code, I will close it for now.